composable-tasks/src/compose_tasks.rs

```diff
@@ -161,6 +166,102 @@ pub trait TaskCall:
         EgldOrEsdtTokenPayment::from(payment_out)
     }
 
+    fn smart_swap(
+        &self,
+        payment_for_current_task: EgldOrEsdtTokenPayment,
+        payments_to_return: &mut PaymentsVec<Self::Api>,
+        args: &mut ManagedVec<ManagedBuffer>,
+    ) -> EgldOrEsdtTokenPayment {
+        require!(
+            !payment_for_current_task.token_identifier.is_egld(),
+            "EGLD can't be swapped!"
+        );
+        let mut payment_in = payment_for_current_task.unwrap_esdt();
```

Comment:
```suggestion
        require!(
            !payment_for_current_task.token_identifier.is_egld(),
            "EGLD can't be swapped!"
        );
        
        let mut payment_in = payment_for_current_task.unwrap_esdt();
```

Personal preference, but I prefer having requires separate from actual code in the function.